layer/layer_store.go

```diff
@@ -35,6 +37,24 @@ type layerStore struct {
 	mountL sync.Mutex
 }
 
+// NewStoreFromConfig creates a new Store instance
+func NewStoreFromConfig(root string, driverName string, graphOptions []string, uidMaps, gidMaps []idtools.IDMap) (Store, error) {
```

Comment:
This could be cleaned up by taking in an options struct and be named `NewStoreWithOptions`

I would image something like
```
type StoreOptions struct {
GraphDriver string
GraphDriverOptions []string
UIDMaps []idtools.IDMap
GIDMaps []idtools.IDMap
}
```

Likewise `NewStore` would maybe renamed `NewStoreFromGraphDriver`, wdyt @tonistiigi ?